plugins/source/gcp/resources/services/iam/roles_fetch.go

```diff
@@ -17,7 +17,7 @@ func fetchRoles(ctx context.Context, meta schema.ClientMeta, r *schema.Resource,
 		return err
 	}
 	for {
-		output, err := iamClient.Projects.Roles.List(c.ProjectId).PageSize(1000).PageToken(nextPageToken).Do()
+		output, err := iamClient.Projects.Roles.List("projects/" + c.ProjectId).PageSize(1000).PageToken(nextPageToken).Do()
```

Comment:
this works, too, but maybe we want to add `ProjectScope` func to `client` that will return `"projects/" + c.ProjectId` and use it? It this the only place this fix is required?

Reply:
> It this the only place this fix is required?

This is the only resources that errors out right out of the box on our playground account with a 404. So it's the only one I could find that needs it.

The format differs between resources, some only need the `ProjectId`, and this is inlined at the code gen level, so I'm not sure there's a reason to put it under the client struct